Signal/src/ViewControllers/InboxTableViewCell.m

```diff
@@ -49,41 +50,56 @@ - (nullable NSString *)reuseIdentifier
     return NSStringFromClass(self.class);
 }
 
-- (void)configureWithThread:(TSThread *)thread contactsManager:(OWSContactsManager *)contactsManager
+- (void)configureWithThread:(TSThread *)thread
+            contactsManager:(OWSContactsManager *)contactsManager
+      blockedPhoneNumberSet:(NSSet<NSString *> *)blockedPhoneNumberSet;
 {
-    if (!_threadId || ![_threadId isEqualToString:thread.uniqueId]) {
-        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
The home has been configuring its cells asynchronously.  This is something we'd like to avoid if possible because:

* It is likely to cause bad frames.
* It wasn't being done safely since cells can be reused.
* Most of the cells configuration should be cheap.

While doing this work, I discovered two other issues:

* We're not caching avatars.  That should be straightforward, and should be a significant perf boost.  For example we reload and configure the home view cells four times on app launch.  
* We're not persisting contact state, so we have a very perceptible "bad frame" on launch where we briefly show contacts as phone numbers.